v2/internal/controllers/insights_workbook_crud_v20230601_test.go

```diff
@@ -0,0 +1,104 @@
+/*
+Copyright (c) Microsoft Corporation.
+Licensed under the MIT license.
+*/
+
+package controllers_test
+
+import (
+	"testing"
+
+	. "github.com/onsi/gomega"
+
+	insights "github.com/Azure/azure-service-operator/v2/api/insights/v1api20230601"
+	"github.com/Azure/azure-service-operator/v2/internal/testcommon"
+	"github.com/Azure/azure-service-operator/v2/internal/util/to"
+)
+
+func Test_Insights_Workbook_v20230601_CRUD(t *testing.T) {
+	t.Parallel()
+
+	tc := globalTestContext.ForTest(t)
+
+	rg := tc.CreateTestResourceGroupAndWait()
+
+	// Create a workbook with the required GUID-based azureName
+	workbookName := tc.Namer.GenerateName("workbook")
+	workbookAzureName := "11111111-2222-3333-4444-555555555555" // GUID format required by Azure
```

Comment:
IIRC we have a helper for generating a "stable" GUID at `tc.Namer.GenerateUUID`.

We also could consider adding a helper for this resource like we have for the SQLRoleAssignment where ASO autogenerates the UUID for the user based on the Kubernetes name/namespace, to save the user the need of doing it themselves - wdyt?